sympy/matrices/expressions/transpose.py

```diff
@@ -37,6 +37,8 @@ def doit(self, **hints):
         if hints.get('deep', True) and isinstance(arg, Basic):
             arg = arg.doit(**hints)
         try:
+            if isinstance(arg, MatrixExpr) and arg.shape == (1, 1):
+                return arg
```

Comment:
@jksuom, looking into the assumptions now ... Besides the 1x1 matrices, it also should handle the case when `A` is square and then `A + A.T` is symmetric. More general case `A + B + A.T + B.T`. 

What would you suggest for the general case? Group all expressions by unique id and check that every expr has a sum with its transposition?